src/SFA.DAS.EmployerAccounts/Services/CommitmentsV2Service.cs

```diff
@@ -0,0 +1,69 @@
+﻿using AutoMapper;
+using SFA.DAS.CommitmentsV2.Api.Client;
+using SFA.DAS.CommitmentsV2.Api.Types.Responses;
+using SFA.DAS.CommitmentsV2.Types;
+using SFA.DAS.CommitmentsV2.Types.Dtos;
+using SFA.DAS.EmployerAccounts.Interfaces;
+using SFA.DAS.EmployerAccounts.Models.Commitments;
+using SFA.DAS.Encoding;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerAccounts.Services
+{
+    public class CommitmentsV2Service : ICommitmentV2Service
+    {
+        private readonly ICommitmentsApiClient _commitmentsApiClient;
+        private readonly IMapper _mapper;
+        private readonly IEncodingService _encodingService;
+
+        public CommitmentsV2Service(ICommitmentsApiClient commitmentsApiClient, IMapper mapper, IEncodingService encodingService)
+        {
+            _commitmentsApiClient = commitmentsApiClient;
+            _mapper = mapper;
+            _encodingService = encodingService;
+        }
+
+        public async Task<IEnumerable<Apprenticeship>> GetDraftApprenticeships(long cohortId)
+        {
+            var draftApprenticeshipsResponse = await _commitmentsApiClient.GetDraftApprenticeships(cohortId);
+
+            return _mapper.Map<IEnumerable<DraftApprenticeshipDto>, List<Apprenticeship>>(draftApprenticeshipsResponse.DraftApprenticeships,
+                opt =>
+                {
+                    opt.AfterMap((src, dest) =>
+                    {
+                        foreach (var apprenticeship in dest)
+                        {
+                            apprenticeship.HashedId = _encodingService.Encode(apprenticeship.Id, EncodingType.ApprenticeshipId);
```

Comment:
Didn't we agree to change this to add a method on the Apprenticeship to Set the hashed id ?

opt.AfterMap((src, dest) => dest.toList().ForEach( c=> c.SetHashId(_encodingService)) ?